src/migrations/1733808052981-item-tags-to-tag.ts

```diff
@@ -0,0 +1,134 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class Migrations1733808052981 implements MigrationInterface {
+  name = 'item-tags-to-tag-1733808052981';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    // insert distinct tags
+    await queryRunner.query(`
+        insert into tag (name, category)
+        select DISTINCT(
+            trim(
+                -- replace faulty breaking lines into spaces
+                replace(
+                    unnest(
+                        -- break comma-separated tags into multiple tags 
+                        string_to_array(
+                            -- remove resulting quotes (") 
+                            substr(
+                                jsonb_array_elements(settings::jsonb->'tags')::text,
+                                2,
+                                length(jsonb_array_elements( settings::jsonb->'tags' )::text) - 2
+                            ),','
```

Comment:
Is there a difference when using `->>` and `->` with json fields ? maybe one or the other will return the value without quotes ?

Edit:  Indeed there is https://arc.net/l/quote/fdgxpxhw

TLDR: `->` returns JSON (so strings will be `"string"`) and `->>` returns TEXT (so strings will be `string` without quotations)

So I think you should be able to get the direct text value with the `->>` operator and simplify your query.

Reply:
I tried but actually `->>` returns a `TEXT` which is not an array. Furthermore it has the following syntax "["a","b"]" which is not the native syntax for array in sql. So I think I better keep my current implementation, as I will anyway have to deal with quotes anyway.

Reply:
Ah... yes if you are dealing with an array as the value, then my suggestion does not make a lot of sens. I am sorry. Looks good like this.